openedx/openedx-platform
# [ENT-327] Add explicit SESSION_EXPIRE_AT_BROWSER_CLOSE setting; default to False

In conjunction with edx/configuration#3820, this pull request enables a setting allowing sessions to be terminated by default on browser close. In general, this should have no implication for normal users, since the login view explicitly sets the session expiry. This will affect SSO users where an explicit expiry time could not be derived from the SSO pipeline (such as when using SAML); setting this variable to True will, in those cases, result in a session created by an SSO login to expire when the user's browser closes.

**JIRA tickets**: Implements [ENT-327](https://openedx.atlassian.net/browse/ENT-327)

**Dependencies**: Soft dependency on edx/configuration#3820

**Screenshots**: N/A; no UI changes

**Sandbox URL**: TBD - sandbox is being provisioned.

**Merge deadline**: ASAP

**Testing instructions**:

1. Configure a SAML provider.
2. With `SESSION_EXPIRE_AT_BROWSER_CLOSE` set to `false` in `lms.env.json`, log in with the SSO provider.
3. Quit your browser entirely.
4. Reopen your browser and navigate to the LMS, noting that you're still logged in.
5. Log out.
6. Switch `SESSION_EXPIRE_AT_BROWSER_CLOSE` to `true` in `lms.env.json` and restart the server.
7. Log in with the SSO provider.
8. Quit your browser entirely.
9. Reopen your browser and navigate to the LMS; note that you're no longer logged in.
10. Log in using a password, leaving the "remember me" box unchecked.
11. Quit your browser entirely.
12. Reopen your browser and navigate to the LMS; note that you're no longer logged in.
13. Log in using a password, checking the "remember me" box.
14. Quit your browser entirely.
15. Reopen your browser and navigate to the LMS; note that you're no logged in.

**Settings**
```yaml
EDXAPP_FEATURES:
  ENABLE_COMBINED_LOGIN_REGISTRATION: true
  ENABLE_THIRD_PARTY_AUTH: true
EDXAPP_SESSION_EXPIRE_AT_BROWSER_CLOSE: true
edx_ansible_source_repo: "https://github.com/open-craft/configuration.git"
configuration_version: "haikuginger/configure-session-expiry"
```
